### ISSUE TITLE
Add stats to app details page

### ISSUE THREAD

**Issue by jgarciao** [jgarciao/linux-store-frontend#19]

**Comment by allanday** [comment]
I agree that this would be good to have, but I wouldn't go too wild. Perhaps just a total number of downloads, or something.

**Comment by ahayzen** [comment]
FYI as some of the graphs I was looking at doing might not go onto flathub.org, i added support in my tool to generate them. I have made a simple page which shows them here https://ahayzen.com/direct/flathub.html  Let me know if you do want any of them or have suggestions, will be happy to help :-)

I personally think that having "Downloads in the past N days" is also useful (maybe we need both) - as if an app had all its downloads in 1 month last year, it is not useful to know that it was downloaded 1000 times last year. I guess that users want to know that this app is current and other people are downloading it. (But then the total downloads is useful for an app author)

Wonder which other info is useful to show ? Maybe some overall graphs are will become interesting overtime ...

**Comment by jgarciao** [comment]
I was thinking of adding a couple of fields in the _Additional information_ section:
- **Installs:** sum of all installs for all the architectures since the app was first published (or since we have stats)
- **Installs (current version)** : Installs/updates for all the architectures since current version was published in the store 

Instead of detailed numbers we can use ranges like in google play (0-100, 100-1000, 1000-10.000, ...)

A couple of interesting related articles:
- [What is the meaning of 1,000,000-5,000,000 installs in google play?](https://android.stackexchange.com/questions/137417/what-is-the-meaning-of-1-000-000-5-000-000-installs-in-google-play)
- [Discrepancies - How Does AppsFlyer Count Installs Compared To App Store and Google Play?](https://support.appsflyer.com/hc/en-us/articles/207040726-Discrepancies-How-Does-AppsFlyer-Count-Installs-Compared-To-App-Store-and-Google-Play-)

**Comment by allanday** [comment]
What's the reason for showing the number of installs for the current version, @jgarciao ?

**Comment by TingPing** [comment]
For what its worth the Snapcraft dashboard shows number of active installs by version which I do believe is useful. One problem is that our stats have no clue really how many of those are active installs or not.

**Comment by allanday** [comment]
> It can give you a clue about the number of active users

I'm not sure that will be obvious to regular users. It does seem like useful information if you're the developer of the app though.

**Comment by TingPing** [comment]
@ahayzen Yes the dev only dashboard.

I'm guessing @jgarciao doesn't want to start adding accounts to the website though.

**Comment by allanday** [comment]
Automated emails might be one alternative for providing feedback to developers.

**Comment by paulcarroty** [comment]
I wrote the simple aggregator - https://github.com/paulcarroty/flathub-app-stats which gives simple y-m-w-d stats for each app. His results can be easily parsed from Flathub frontend.
Statistics are updated daily with TravisCI [worker](https://github.com/paulcarroty/flathub-app-stats/blob/master/.travis.yml).

Ping @alexlarsson @ahayzen

**Comment by jurf** [comment]
@paulcarroty what happened to that?

**Comment by paulcarroty** [comment]
@jurf the downloads of every app was calculated & sorted for 4 periods.

**Comment by jurf** [comment]
Oh, I thought that would be daily.

**Comment by paulcarroty** [comment]
If you/someone are interested to integrate it to the store frontend, the time format can be easily changed.

**Comment by jgarciao** [comment]
Hi Paul,

Firt of all thanks for your contribution. I've been away from keyboard most of August and now I'm slowly starting to catch up with the development of the website (I'm doing this in my spare time)

My idea was to implement something similar to your code in linux-store-backend and store the results in the database. I also want to have ostree commit information for each app/runtime in order to have downloads by commit to try to better guess active users per app

I don't have any estimation when this will be available

**Comment by BrunoVernay** [comment]
It would be nice to be able to filter by "Last update" 1 year / 2 year ...   and by number of downloads too.
I think it relates to this thread, but I could add an new issue if needed.

**Comment by razzeee** [comment]
Added a basic version to the new frontend https://flathub.vercel.app/apps/details/com.valvesoftware.Steam

**Comment by jurf** [comment]
I agree with @aday, this might be confusing for regular users (great for developers though). A total installs counter might be a good conservative start.

**Comment by fedelibre** [comment]
@razzeee How can I check the number of downloads *per architecture*? I know this is not useful for users, but as a maintainer I'm interested to know the number of downloads of aarch64 packages.

**Comment by razzeee** [comment]
I can adjust the logic, to keep that data around and display it. It might not be a good idea for the normal flathub detail page, but we could start with that.

The data is available on https://flathub.org/stats/2021/06/06.json for e.g.

**Comment by razzeee** [comment]
Having had a look (don't mind the colors) it doesn't seem to make sense, to add this to the user facing page. With that chart height and low downloads of all but one platform, even a stacked bar chart becomes unusable.

The yellow here for e.g. are aarch64 downloads.

![image](https://user-images.githubusercontent.com/5943908/146102355-00fab957-2b82-408c-bbcd-b9c05a5ab314.png)

**Comment by BrunoVernay** [comment]
A per year granularity would be enough and more readable.  
Having the 4 architectures does not look like a problem to me, it just show the reality, at least for this package.

**Comment by razzeee** [comment]
Haven't found any package that's different, most are in deed worse, I've picked one with not as many downloads in general for the above.

Makes me wonder, if a pie chart with just all downloads by arch would be more usefull, but you won't be able to see trends there.

**Comment by real-yfprojects** [comment]
> Having had a look (don't mind the colors) it doesn't seem to make sense, to add this to the user facing page. With that chart height and low downloads of all but one platform, even a stacked bar chart becomes unusable.
> 
> The yellow here for e.g. are aarch64 downloads.
> 
> ![image](https://user-images.githubusercontent.com/5943908/146102355-00fab957-2b82-408c-bbcd-b9c05a5ab314.png)

I suggest aggregating the downloads per month in that chart. Then it only has a single data point per month.

**Comment by BrunoVernay** [comment]
Good idea and the sum over the last 6 months could be used to sort by number of downloads.
And already think about a way to prevent or at least alert  in case this system is abused to artificially increase the number of downloads to trick people into installing a bad package

**Comment by real-yfprojects** [comment]
> Good idea and the sum over the last 6 months could be used to sort by number of downloads.

How are the apps for `most popular` currently calculated?

> And already think about a way to prevent or at least alert in case this system is abused to artificially increase the number of downloads to trick people into installing a bad package

That's one of the reasons why other package distribution sites like [pypi.org](https://packaging.python.org/en/latest/guides/analyzing-pypi-package-downloads/#background) do not provide stats in their web GUI. However npmjs.com do show weekly download stats for the last year.
